File: 009/main.js
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
  let str = x + ''
  const len = str.length
  let flag = true
  if (len % 2 === 0) {
    for (let i = 0; i < len / 2; i++) {
      if (str[len / 2 - i - 1] !== str[len / 2 + i]) {
        flag = false
        break
      }
    }
  } else {
    for (let i = 1; i < len / 2; i++) {
      if (str[(len - 1) / 2 - i] !== str[(len - 1) / 2 + i]) {
        flag = false
        break
      }
    }
  }
  return flag
}

console.log(isPalindrome(1221))
console.log(isPalindrome(1231))
console.log(isPalindrome(121))
console.log(isPalindrome(-2147483648))

File: 009/main.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    bool isPalindrome(int x) {
        if (x == 0) return true;
        if (x < 0) return false;
        vector<int> arr;
        int mid;
        while (x) {
            mid = x % 10;
            arr.push_back(mid);
            x = x / 10;
        }
        bool flag = true;
        int len = arr.size();
        if (len % 2 == 0) {
            for (int i = 0; i < len / 2; i++) {
                if (arr[len / 2 - i - 1] != arr[len / 2 + i]) {
                    flag = false;
                    break;
                }
            }
        } else {
            for (int i = 0; i < len / 2 + 1; i++) {
                if (arr[len / 2 - i] != arr[len / 2 + i]) {
                    flag = false;
                    break;
                }
            }
        }
        return flag;
    }
};

int main()
{
    Solution sol;
    bool res = sol.isPalindrome(100);
    cout << res << endl;
    return 0;
}
